### PR TITLE
Add a Skeleton for Collector

### PR DESCRIPTION
**Pull request by pjanotti**
Keeping the pace of small changes to easy up on the reviews.

This one adds the skeleton for collector, right now it runs but just logs
the received spans.

This introduces a proper logger to the project, zap, while it is not of practical
use a this stage a logger is a need for a service like the collector.

### CODE REVIEW

**Comment by yurishkuro** [comment]
all I see is a makefile change, where's the collector itself?

**Comment by pjanotti** [comment]
Ops, forgot adding the main.go... sorry about that.

**Comment by pjanotti** [comment]
Test failure doesn't seem related to this change:
```
--- FAIL: TestConversionRoundtrip (0.21s)
	trace_interceptor_test.go:390: The roundtrip JSON doesn't match the JSON that we want
		Got:
		[{"timestamp":1472470996199000,"duration":207000,"traceId":"4d1e00c0db9010db86154a4ba6e91385","id":"4d1e00c0db9010db","parentId":"86154a4ba6e91385","name":"get","kind":"CLIENT","localEndpoint":{"serviceName":"frontend","ipv6":"7::80:807f"},"annotations":[{"timestamp":1472470996238000,"value":"foo"},{"timestamp":1472470996403000,"value":"bar"}],"tags":{"clnt/finagle.version":"6.45.0","http.path":"/api"}},{"timestamp":1472470996199000,"duration":207000,"traceId":"4d1e00c0db9010db86154a4ba6e91385","id":"4d1e00c0db9010db","parentId":"86154a4ba6e91386","name":"put","kind":"SERVER","localEndpoint":{"serviceName":"frontend","ipv6":"7::80:807f"},"annotations":[{"timestamp":1472470996238000,"value":"foo"},{"timestamp":1472470996403000,"value":"bar"}],"tags":{"clnt/finagle.version":"6.45.0","http.path":"/api"}}]
		Want:
		[{"traceId":"4d1e00c0db9010db86154a4ba6e91385","parentId":"86154a4ba6e91385","id":"4d1e00c0db9010db","kind":"CLIENT","name":"get","timestamp":1472470996199000,"duration":207000,"localEndpoint":{"serviceName":"frontend","ipv4":"0.0.0.0","ipv6":"7::80:807f"},"annotations":[{"timestamp":1472470996238000,"value":"foo"},{"timestamp":1472470996403000,"value":"bar"}],"tags":{"http.path":"/api","clnt/finagle.version":"6.45.0"}},{"traceId":"4d1e00c0db9010db86154a4ba6e91385","parentId":"86154a4ba6e91386","id":"4d1e00c0db9010db","kind":"SERVER","name":"put","timestamp":1472470996199000,"duration":207000,"localEndpoint":{"serviceName":"frontend","ipv4":"0.0.0.0","ipv6":"7::80:807f"},"annotations":[{"timestamp":1472470996238000,"value":"foo"},{"timestamp":1472470996403000,"value":"bar"}],"tags":{"http.path":"/api","clnt/finagle.version":"6.45.0"}}]
FAIL
```

**Comment by pjanotti** [comment]
The test failure is caused by https://github.com/openzipkin/zipkin-go/issues/85 related to https://github.com/census-instrumentation/opencensus-service/pull/145

**Comment by pjanotti** [comment]
Tests are passing now after #145 was merged.

File: cmd/occollector/main.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+	// TODO: orderly shutdown: first receivers, then flushing pipelines giving
+	// senders a chance to send all their data. This may take time, the allowed
+	// time should be part of configuration.
+	closeSrv()
+
+	logger.Info("Shutdown complete.")
+}
+
+func runOCServerWithInterceptor(addr string, logger *zap.Logger) (func() error, error) {
```
Comment (yurishkuro) [inline cmd/occollector/main.go]: I'd recommend everything from this line and below to be in a sub-package, with unit tests. Keep main minimal.

Comment (odeke-em) [reply cmd/occollector/main.go]: @yurishkuro thanks for the suggestion. However that code is already unit tested
* https://github.com/census-instrumentation/opencensus-service/blob/f88f2995084f1dcbb5aa9466081d953d0d23e5e4/interceptor/opencensus/observability_test.go#L46
* https://github.com/census-instrumentation/opencensus-service/blob/f88f2995084f1dcbb5aa9466081d953d0d23e5e4/interceptor/opencensus/opencensus_test.go#L46
et al and follows with `cmd/ocagent`

Comment (pjanotti) [reply cmd/occollector/main.go]: Thanks @yurishkuro, as the project grows we will do that - some refactoring on packages and separation of concerns is needed as the project grows.

**Comment by pjanotti** [comment]
Thanks for reviewing @yurishkuro and @odeke-em

**Pull request merged**